sysboundary/ionosphere.cpp

```diff
@@ -613,14 +591,14 @@ namespace SBC {
       // Copy each face B-field from the cell on the other side of it
       switch(component) {
```

Comment:
This block needs to be fixed so it copies from the simulation cell direction instead of just the -1 direction, such as:


// Copy each face B-field from the cell in the simulation cell direction
switch(component) {
	case 0:
		if (technicalGrid.get(i-1,j,k)->sysBoundaryLayer==1) {
			return bGrid->get(i-1,j,k)->at(fsgrids::bfield::PERBX + component);
		} else { // Is sysboundarylayer 2, read from opposite direction
			return bGrid->get(i+1,j,k)->at(fsgrids::bfield::PERBX + component);
		}
	case 1:
		if (technicalGrid.get(i,j-1,k)->sysBoundaryLayer==1) {
			return bGrid->get(i,j-1,k)->at(fsgrids::bfield::PERBX + component);
		} else { // Is sysboundarylayer 2, read from opposite direction
			return bGrid->get(i,j+1,k)->at(fsgrids::bfield::PERBX + component);
		}
	case 2:
		if (technicalGrid.get(i,j,k-1)->sysBoundaryLayer==1) {
			return bGrid->get(i,j,k-1)->at(fsgrids::bfield::PERBX + component);
		} else { // Is sysboundarylayer 2, read from opposite direction
			return bGrid->get(i,j,k+1)->at(fsgrids::bfield::PERBX + component);
		}
         default:
		cerr << "ERROR: ionosphere boundary tried to copy nonsensical magnetic field component " << component << endl;
		return 0.0;
      }

Reply:
Hmm but this will segfault if getting a -1 neighbour from a cell at coordinate 0. But I'll make it safe somehow.

Reply:
Doh we're in the ionosphere. But this means we likely break the support of having outflow crossing the ionosphere. Which we rarely used, though.

Reply:
Right, and the outflow doesn't explicitly call a -1 or +1 neighbour, but instead copies from the closest internal cell.

Reply:
That's implemented now.